lib/msal-browser/src/cache/AsyncMemoryStorage.ts

```diff
@@ -112,12 +116,34 @@ export class AsyncMemoryStorage<T> implements IAsyncStorage<T> {
     /**
      * Clears in-memory Map and tries to delete the IndexedDB database.
      */
-    async clear(): Promise<void> {
+    async clear(): Promise<boolean> {
+    // InMemory cache is a Map instance, clear is straightforward
+        this.logger.verbose(`Deleting in-memory keystore ${this.logStoreName()}`);
         this.inMemoryCache.clear();
+        this.logger.verbose(`In-memory keystore ${this.logStoreName()} deleted`);
+        this.logger.verbose(`Deleting persistent keystore ${this.logStoreName()}`);
+        
         try {
-            await this.indexedDBCache.deleteDatabase();
+            const dbDeleted = await this.indexedDBCache.deleteDatabase();
+            
+            if (dbDeleted) {
+                this.logger.verbose(`Persistent keystore ${this.logStoreName()} deleted`);
+            }
+            
+            return dbDeleted;
         } catch (e) {
-            this.handleDatabaseAccessError(e);
+            if (e instanceof Error) {
+                this.handleDatabaseAccessError(e);
+            }
+            return false;
         }
     }
+
+    /**
+     * Helper that prints the keystore name in a log-friendly way
+     * @returns 
+     */
+    private logStoreName() {
+        return (this.name) ? `"${this.name}"` : Constants.EMPTY_STRING;
```

Comment:
Did you mean to include the quotes in teh message?

Reply:
if not, could also be:

```suggestion
        return (this.name) || Constants.EMPTY_STRING;
```

Reply:
I did since it's meant to indicate the given name of the key store... Do we prefer to not make the distinction?

Basically: Deleting keystore "asymmetricKeys" / Keystore "asymmetricKeys" deleted

vs

Deleting asymmetricKeys keystore / Keystore asymmetricKeys deleted

Reply:
Yeah not a big deal to me, log message are for us and we know the intent here.

Reply:
Since we're going with a required storeName attribute, this method is no longer necessary, replacing with the regular property accessor everywhere it's called.